backend/node/genvm/base.py

```diff
@@ -204,14 +203,67 @@ def get_contract_schema(contract_code: str) -> dict:
                             ):
                                 variables[stmt.target.id] = stmt.annotation.id
 
+            abi = GenVM.generate_abi_from_schema_methods(methods)
+
             contract_schema = {
                 "class": class_name,
                 "methods": methods,
                 "variables": variables,
+                "abi": abi,
             }
 
         return contract_schema
 
+    @staticmethod
+    def get_abi_param_type(param_type: str) -> str:
```

Comment:
Now we also generate an Ethereum-compatible ABI for the intelligent contract interface.
The current way of doing it (https://github.com/yeagerai/genlayer-simulator/pull/315/files#diff-f9c09ce67d4e8dddb05fbad6e64f19e0462222f8ab24310cc8edaa016f32e7a5R157) will be deprecated when the frontend gets refactored.

Reply:
Let's address the ABI creation later when the GenVM and the Node make some progress.